apps/server/src/modules/tool/context-external-tool/service/tool-version-service.ts

```diff
@@ -0,0 +1,44 @@
+import { Injectable } from '@nestjs/common/decorators/core/injectable.decorator';
+import { Inject } from '@nestjs/common';
+import { ContextExternalToolValidationService } from './context-external-tool-validation.service';
+import { SchoolExternalToolValidationService } from '../../school-external-tool/service';
+import { IToolFeatures, ToolFeatures } from '../../tool-config';
+import { ExternalTool } from '../../external-tool/domain';
+import { SchoolExternalTool } from '../../school-external-tool/domain';
+import { ContextExternalTool } from '../domain';
+import { ToolConfigurationStatus } from '../../common/enum';
+import { CommonToolService } from '../../common/service';
+
+@Injectable()
+export class ToolVersionService {
+	constructor(
+		private readonly contextExternalToolValidationService: ContextExternalToolValidationService,
+		private readonly schoolExternalToolValidationService: SchoolExternalToolValidationService,
+		private readonly commonToolService: CommonToolService,
+		@Inject(ToolFeatures) private readonly toolFeatures: IToolFeatures
+	) {}
+
+	async determineToolConfigurationStatus(
+		externalTool: ExternalTool,
+		schoolExternalTool: SchoolExternalTool,
+		contextExternalTool: ContextExternalTool
+	): Promise<ToolConfigurationStatus> {
+		// TODO N21-1337 remove if statement, when feature flag is removed
+		if (this.toolFeatures.toolStatusWithoutVersions) {
+			try {
+				await this.schoolExternalToolValidationService.validate(schoolExternalTool);
+				await this.contextExternalToolValidationService.validate(contextExternalTool);
+				return ToolConfigurationStatus.LATEST;
+			} catch (err) {
+				return ToolConfigurationStatus.OUTDATED;
+			}
+		}
+		const status: ToolConfigurationStatus = this.commonToolService.determineToolConfigurationStatus(
```

Comment:
Can you use a different name for this function or the function in commonToolService?
Are they doing the same or why they have the same name...

Reply:
they do, just in another ways. commonToolService uses versions, which will be legacy and can be removed in the future

Reply:
Then please deprecate the commonToolService with a hint what is to you used instead